pkg/driver/builder.go

```diff
@@ -0,0 +1,90 @@
+package driver
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/LambdaTest/test-at-scale/pkg/core"
+	"github.com/LambdaTest/test-at-scale/pkg/errs"
+	"github.com/LambdaTest/test-at-scale/pkg/global"
+	"github.com/LambdaTest/test-at-scale/pkg/lumber"
+)
+
+type (
+	Builder struct {
+		Logger               lumber.Logger
+		TestExecutionService core.TestExecutionService
+		TestDiscoveryService core.TestDiscoveryService
+		AzureClient          core.AzureClient
+		BlockTestService     core.BlockTestService
+		ExecutionManager     core.ExecutionManager
+		TASConfigManager     core.TASConfigManager
+		CacheStore           core.CacheStore
+		DiffManager          core.DiffManager
+		ListSubModuleService core.ListSubModuleService
+	}
+	NodeInstaller struct {
+		logger           lumber.Logger
+		ExecutionManager core.ExecutionManager
+	}
+)
+
+func (b *Builder) GetDriver(version int) (core.Driver, error) {
+	if version < global.NewTASVersion {
```

Comment:
should be switch-case ideally, erring out on default (like if someone provides version as 3). If someone is NOT providing the version, let's have a constant defined to fallback (essentially the least backported version). Benefit of this - in future, we can stop supporting few versions and forcefully fail repositories if backporting becomes messy.